### ISSUE TITLE
Drones sink and drown

### ISSUE THREAD

**Issue by Haze33E** [suppergerrie2/ModJam5#15]
Mod version: 1.3.0

Forge version: 14.23.4.2705 

Java version: 8 (build 1.8.0_181-b13)

Minecraft version: 1.12

What happened? 
Drones sink and drown

What did you expect to happen? 
for them to swim and jump out of water

How to reproduce?
either place a drone in water or allow them to path into it themselves

**Comment by suppergerrie2** [comment]
They are supposed to die (Except the crop farm drone) in water, but should not path into it.

**Comment by suppergerrie2** [comment]
As far as I can see the drones don't go in water unless pushed or a block is replaced beneath their feet. So this is working as intended.

**Comment by suppergerrie2** [comment]
Well nevermind, I just had a hauler drone taking a bath...

**Comment by Haze33E** [comment]
Yeah I was experimenting with stacking crop farms with the drones and had an open water hole for a farm and a hauler jumped in and sat their until it drowned.

**Comment by Haze33E** [comment]
Oh yeah i forgot to mention the drones won't travel over minecart rails or magma blocks. I noticed mobs have the same problem. The magma blocks makes sense since the drone would die. But the rails might cause problems.

**Comment by suppergerrie2** [comment]
That's vanilla behaviour, probably added to make railways less annoying.

**Comment by suppergerrie2** [comment]
So I am not sure what to do. I can keep it like this and adding a "swim upgrade". Or make them able to swim automatically. When I started working on the mod I thought it was a nice idea to make them unable to swim, but that seems more annoying than a challenge.

Thinking about it, I could also make them like iron golems. Not being able to swim but just walking underwater. They don't need air and are a small cube of iron so it kinda makes sense? What do you think?

**Comment by Haze33E** [comment]
It would be nice if they didn't drown. But if they're sitting in water only one block deep it would be nice if they jumped out as well. If it's 2 blocks deep then yeah it makes sense they would get stuck.
